okhttp/third_party/okhttp/main/java/io/grpc/okhttp/internal/framed/Hpack.java

```diff
@@ -373,26 +389,105 @@ ByteString readByteString() throws IOException {
 
   static final class Writer {
     private final Buffer out;
+    private boolean useCompression;
+    // Visible for testing.
+    int headerTableSizeSetting;
+
+    /**
+     * In the scenario where the dynamic table size changes multiple times between transmission of
+     * header blocks, we need to keep track of the smallest value in that interval.
+     */
+    private int smallestHeaderTableSizeSetting = Integer.MAX_VALUE;
+    private boolean emitDynamicTableSizeUpdate = false;
+    private int maxDynamicTableByteCount;
+
+    // Visible for testing.
+    io.grpc.okhttp.internal.framed.Header[] dynamicTable = new io.grpc.okhttp.internal.framed.Header[8];
+    // Array is populated back to front, so new entries always have lowest index.
+    int nextDynamicTableIndex = dynamicTable.length - 1;
+    int dynamicTableHeaderCount = 0;
```

Comment:
nit: this field is private and has a package-private/VisibleForTesting getter, that is, can not be mutated outside the class.

Reply:
Done.